paddle/fluid/framework/tensor.h

```diff
@@ -26,14 +26,21 @@ limitations under the License. */
 #include "paddle/fluid/platform/device_context.h"
 #include "paddle/fluid/platform/enforce.h"
 #include "paddle/fluid/platform/place.h"
+#ifdef PADDLE_WITH_MKLDNN
+#include "paddle/fluid/framework/mkldnn_tensor.h"
+#endif
 
 namespace paddle {
 
 namespace framework {
 
 class LoDTensor;
 
-class Tensor {
+class Tensor
+#ifdef PADDLE_WITH_MKLDNN
+    : public MKLDNNTensor
```

Comment:
I think it's not a good way let `Tensor : public MKLDNNTensor`

Reply:
May be
```
class Tensor {
#ifdef PADDLE_WITH_MKLDNN
  mkl_fileds;
  void mkldnn_method();
#endif
};
```
is a better way

Reply:
@jacquesqiao and @tensor-tang, The inheritance was removed from the code, and the all methods was moved from `mkldnn_tensor` to `tensor` file